py/client/pydeephaven/agg.py

```diff
@@ -300,17 +313,55 @@ def std(cols: Union[str, List[str]] = None) -> Aggregation:
     return _AggregationColumns(agg_spec=agg_spec, cols=to_list(cols))
 
 
-def unique(cols: Union[str, List[str]] = None) -> Aggregation:
-    """Creates a Unique aggregation.
+def unique(cols: Union[str, List[str]] = None, include_nulls: bool = False,
+           non_unique_sentinel: Union[np.number, str, bool] = None) -> Aggregation:
+    """Creates a Unique aggregation which computes the single unique value within an aggregation group for each of
+    the given columns. If all values in a column are null, or if there is more than one distinct value in a column, the
+    result is the specified non_unique_sentinel value (defaults to null).
 
     Args:
         cols (Union[str, List[str]]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
             default is None, only valid when used in Table agg_all_by operation
+        include_nulls (bool): whether null is treated as a value for the purpose of determining if the values in the
+            aggregation group are unique, default is False.
+        non_unique_sentinel (Union[np.number, str, bool]): the non-null sentinel value when no unique value exists,
+            default is None. Must be a non-None value when include_nulls is True. When passed in as a numpy scalar
+            number value, it must be of one of these types: np.int8, np.int16, np.uint16, np.int32, np.int64, np.float32
+            , np.float64.
+
+    Raises:
+        TypeError
 
     Returns:
         an aggregation
     """
-    agg_spec = _GrpcAggSpec(unique=_GrpcAggSpec.AggSpecUnique(include_nulls=False, non_unique_sentinel=None))
+    if non_unique_sentinel is not None:
+        if isinstance(non_unique_sentinel, np.byte):
+            agg_spec_non_unique_sentinel = _GrpcAggSpec.AggSpecNonUniqueSentinel(byte_value=non_unique_sentinel)
+        elif isinstance(non_unique_sentinel, np.short):
+            agg_spec_non_unique_sentinel = _GrpcAggSpec.AggSpecNonUniqueSentinel(short_value=non_unique_sentinel)
+        elif isinstance(non_unique_sentinel, np.int32):
+            agg_spec_non_unique_sentinel = _GrpcAggSpec.AggSpecNonUniqueSentinel(int_value=non_unique_sentinel)
+        elif isinstance(non_unique_sentinel, np.int64):
+            agg_spec_non_unique_sentinel = _GrpcAggSpec.AggSpecNonUniqueSentinel(long_value=non_unique_sentinel)
+        elif isinstance(non_unique_sentinel, np.float32):
+            agg_spec_non_unique_sentinel = _GrpcAggSpec.AggSpecNonUniqueSentinel(float_value=non_unique_sentinel)
+        elif isinstance(non_unique_sentinel, np.float64):
+            agg_spec_non_unique_sentinel = _GrpcAggSpec.AggSpecNonUniqueSentinel(double_value=non_unique_sentinel)
+        elif isinstance(non_unique_sentinel, np.uint16):
+            agg_spec_non_unique_sentinel = _GrpcAggSpec.AggSpecNonUniqueSentinel(char_value=non_unique_sentinel)
```

Comment:
What is the thinking on supporting `uint16` as `char` instead of going to `int` or just not supporting it?  Similarly, `uint32` could be supported as `long`, but that is a size change that we may want to avoid.  Is it just the closest thing to a char in np?

Reply:
I want to point out that there is a bit of funky usability to consider.  To actually do a char in py, you would do something like `np.uint16(ord("S"))`, which is on the complex side for a simple concept.  Not sure there is a better solution.